# CASE: O vendedor de Pirozhki

Vladimir Sidorov desistiu de seu trabalho convencional na KGB e decidiu empreender em 2010 no ramo alimentício. Com suas economias abriu um foodtruck especializado na venda de Pirozhki, onde seu grande diferencial era trabalhar com ingredientes importados dando uma qualidade extraordinária ao produto. Após o crescimento das vendas, a modalidade delivery foi implementada para auxiliar a empresa. Vladimir era extremamente popular no bairro, todos o conheciam, mas em 2019 se mudou para um ponto melhor e não tivemos mais notícia. Os últimos dados compartilhados das vendas de Vlad foram disponibilizados antes de sua partida.

| Ano  | Vendas |
| ---- | ------ |
| 2010 | 723    |
| 2011 | 814    |
| 2012 | 905    |
| 2014 | 1087   |
| 2017 | 1360   |
| 2018 | 1451   |
| 2019 | 1542   |

1.  Dado o problema acima, qual o melhor algoritmo para descobrir os dados de 2015?
2.  Qual coeficiente angular?
3.  Considerando uma perspectiva estatística, qual o total de venda em 2020 (Apresentar valores)?
4.  Considerando uma perspectiva estatística, qual o total de venda em 2021 (Apresentar valores)?
5.  Desenvolva um código em python que calcule estatisticamente as vendas em qualquer ano desejado
6.  Considerando sua opinião, qual o total de venda em 2021?
7.  Considerando sua opinião, qual o total de venda em 2023?


- Nome: Cauã Simões Pinheiro
- RM: 98319

## Respostas

1. O melhor algoritmo para descobrir os dados de 2015 é o de regressão linear, pois ele é capaz de prever valores futuros com base em dados passados.


In [24]:
# 2. https://pt.wikipedia.org/wiki/Declive
import numpy as np

X = np.array([2010, 2011, 2012, 2014, 2017, 2018, 2019])
y = np.array([723, 814, 905, 1087, 1360, 1451, 1542])

media_X = np.mean(X)
media_y = np.mean(y)

numerador = np.sum((X - media_X) * (y - media_y))
denominador = np.sum((X - media_X) ** 2)

coeficiente_angular = numerador / denominador

coeficiente_angular

91.00000000000001

In [28]:
# 3.

print(f"Total de venda de 2020: {coeficiente_angular * (2020 - media_X) + media_y}")

# ou

y_2019 = y[-1]

y_2019 + coeficiente_angular

# 4.

print(f"Total de venda de 2021: {coeficiente_angular * (2021 - media_X) + media_y}")

# ou

y_2019 + 2 * coeficiente_angular

Total de venda de 2020: 1633.0000000000089
Total de venda de 2021: 1724.000000000009


1724.0

In [31]:
# 5.


def prev_ano(ano):
    return coeficiente_angular * (ano - media_X) + media_y


prev_ano(2023)

1906.000000000009

6.

Sabendo que tivemos uma pandemia em 2020, temos três cenários possíveis:

- Cenário 1: Vladimir se adaptou e começou a vender por aplicativos de entrega, como iFood e Uber Eats.
  Nesse caso, o total de vendas deve continuar a reta do coeficiente, então, o total de vendas em 2021 seria de 1724.
- Cenário 2: Vladimir não se adaptou e continuou vendendo apenas presencialmente. Nesse caso, o total de vendas deve cair drasticamente, então, o total de vendas em 2021 seria menor que 600.
- Cenário 3: Como aconteceu com muitos negócios, Vladimir não se adaptou e acabou fechando o seu negócio. Nesse caso, o total de vendas seria 0.

7.

Seguindo a mesma lógica do item 6, temos três cenários possíveis:

- Cenário 1: Vladimir se adaptou e começou a vender por aplicativos de entrega, como iFood e Uber Eats. Em 2023, ele pode decidir se continua com o modelo de venda por aplicativo ou volta a ser um foodtruck.  
  De qualquer maneira, o total de vendas deve continuar a reta do coeficiente, então, o total de vendas em 2023 seria de 1906.
- Cenário 2: Vladimir não se adaptou e continuou vendendo apenas presencialmente. Nesse caso, o total de vendas deve cair drasticamente e volta a se recuperar somente em 2023, então, o total de vendas em 2023 seria algo perto do que era em 2019, por volta de 1500.
- Cenário 3: Como aconteceu com muitos negócios, Vladimir não se adaptou e acabou fechando o seu negócio na pandemia. Nesse caso, o total de vendas seria 0.
